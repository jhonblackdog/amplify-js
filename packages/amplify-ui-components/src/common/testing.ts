export const pixelThreshold = 10;
