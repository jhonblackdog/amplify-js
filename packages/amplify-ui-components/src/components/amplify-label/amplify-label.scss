.label {
  color: var(--amplify-label-color);
  font-size: 14px;
  margin-bottom: 16px;
}
