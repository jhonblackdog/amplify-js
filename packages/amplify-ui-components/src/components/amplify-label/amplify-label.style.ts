import { css, cx } from 'emotion';

export const label = css`
  color: var(--label-color);
  font-size: 14px;
  margin-bottom: 16px;
`;
