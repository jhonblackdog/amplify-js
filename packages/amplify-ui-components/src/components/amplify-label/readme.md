# amplify-label



<!-- Auto Generated Below -->


## Properties

| Property        | Attribute        | Description | Type      | Default     |
| --------------- | ---------------- | ----------- | --------- | ----------- |
| `htmlFor`       | `html-for`       |             | `string`  | `undefined` |
| `styleOverride` | `style-override` |             | `boolean` | `false`     |


## Dependencies

### Used by

 - [amplify-checkbox](../amplify-checkbox)
 - [amplify-form-field](../amplify-form-field)
 - [amplify-radio-button](../amplify-radio-button)

### Graph
```mermaid
graph TD;
  amplify-checkbox --> amplify-label
  amplify-form-field --> amplify-label
  amplify-radio-button --> amplify-label
  style amplify-label fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
