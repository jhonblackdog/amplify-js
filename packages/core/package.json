{
  "name": "@aws-amplify/core",
  "version": "1.0.26-unstable.1",
  "description": "Core category of aws-amplify",
  "main": "./dist/cjs/index.js",
  "module": "./dist/esm/src/index.js",
  "typings": "./dist/esm/src/index.d.ts",
  "sideEffects": false,
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "test": "tslint 'src/**/*.ts' && jest --coverage",
    "build-with-test": "npm test && npm run build",
    "build:cjs": "tsc -p tsconfig.es5.json && rollup -c",
    "build:esm": "tsc -p tsconfig.es2015.json",
    "build": "npm run clean && npm run build:cjs && npm run build:esm",
    "clean": "rimraf tsc-out dist",
    "format": "tsfmt --useTsfmt tsfmt.json -r src/**/*.ts",
    "lint": "tslint 'src/**/*.ts'"
  },
  "react-native": {
    "./lib/ClientDevice": "./lib/ClientDevice/reactnative.js",
    "./lib/RNComponents": "./lib/RNComponents/reactnative.js",
    "./lib/StorageHelper": "./lib/StorageHelper/reactnative.js"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/aws/aws-amplify.git"
  },
  "author": "Amazon Web Services",
  "license": "Apache-2.0",
  "bugs": {
    "url": "https://github.com/aws/aws-amplify/issues"
  },
  "homepage": "https://github.com/aws/aws-amplify#readme",
  "devDependencies": {
    "find": "^0.2.7",
    "prepend-file": "^1.3.1",
    "prettier": "^1.7.4",
    "typescript": "^3.2.2"
  },
  "dependencies": {
    "aws-sdk": "2.329.0",
    "url": "^0.11.0"
  },
  "jest": {
    "transform": {
      "^.+\\.(js|jsx|ts|tsx)$": "<rootDir>../../node_modules/ts-jest/preprocessor.js"
    },
    "testRegex": "(/__tests__/.*|\\.(test|spec))\\.(tsx?|jsx?)$",
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js",
      "json",
      "jsx"
    ],
    "testEnvironment": "jsdom",
    "mapCoverage": true,
    "coverageThreshold": {
      "global": {
        "branches": 0,
        "functions": 0,
        "lines": 0,
        "statements": 0
      }
    },
    "testURL": "http://localhost/",
    "coveragePathIgnorePatterns": [
      "/node_modules/"
    ]
  }
}
