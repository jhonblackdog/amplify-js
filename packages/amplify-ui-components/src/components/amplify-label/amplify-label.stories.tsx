import { storiesOf } from '@storybook/html';
import { knobs } from '../../common/testing';

storiesOf('amplify-label', module).add('with text', () => {
  const label = knobs.labelKnob('Label text');
  const override = knobs.overrideStyleKnob();
  return `<amplify-label override-style=${override}>${label}</amplify-button>`;
});