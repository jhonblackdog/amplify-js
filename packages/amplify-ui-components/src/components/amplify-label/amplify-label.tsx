import { Component, Prop, h } from '@stencil/core';
import { label } from './amplify-label.style';
import { styleNuker } from '../../common/helpers';
import { AMPLIFY_UI_LABEL } from '../../common/constants';
@Component({
  tag: 'amplify-label',
  shadow: false,
})
export class AmplifyLabel {
  @Prop() htmlFor: string;
  @Prop() styleOverride: boolean = false;

  render() {
    return (
      <label class={styleNuker(this.styleOverride, AMPLIFY_UI_LABEL, label)} htmlFor={this.htmlFor}>
        <slot />
      </label>
    );
  }
}
