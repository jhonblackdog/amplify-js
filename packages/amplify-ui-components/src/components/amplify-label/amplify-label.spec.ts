import { newSpecPage } from '@stencil/core/testing';
import { AmplifyLabel } from './amplify-label';

describe('amplify-label spec:', () => {
  describe('Component logic ->', () => {
    let label;

    beforeEach(() => {
      label = new AmplifyLabel();
    });

    it('should render false for style override prop by default', () => {
      expect(label.overrideStyle).toBe(false);
    });
  });
  describe('Render logic ->', () => {
    it('should render a label by default', async () => {
      const page = await newSpecPage({
        components: [AmplifyLabel],
        html: `<amplify-label></amplify-label>`
      });

      expect(page.root).toMatchSnapshot();
    });

    it('should render only class `amplify-ui--label` when override style is set to true', async () => {
      const page = await newSpecPage({
        components: [AmplifyLabel],
        html: `<amplify-label override-style="true"></amplify-label>`
      });

      expect(page.root).toMatchSnapshot();
    });
  });
});