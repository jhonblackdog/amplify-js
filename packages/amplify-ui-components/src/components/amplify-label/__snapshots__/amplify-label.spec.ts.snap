// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-label spec: Render logic -> should render a label by default 1`] = `
<amplify-label>
  <!---->
  <label class="amplify-ui--label css-uiv44s"></label>
</amplify-label>
`;

exports[`amplify-label spec: Render logic -> should render only class \`amplify-ui--label\` when override style is set to true 1`] = `
<amplify-label override-style="true">
  <!---->
  <label class="amplify-ui--label"></label>
</amplify-label>
`;
